api/python/provisioner/commands/check.py

```diff
@@ -1104,6 +1053,60 @@ def _controller_ip(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
 
         return res
 
+    def _validate_bmc_check(self, check: str):
```

Comment:
As per @dmitrydb suggestion I have not used staticmethod.
 
It is better to make _bmc_accesibility as ordinary class method (remove @staticmethod). So it will allow to refer to _validate_bmc_check using self. and be independent of class renaming:

`return self._validate_bmc_check('accessible')`

Reply:
I think @andkononykhin2 means something different. You can do this: keep this two method as non-static:
```
    def _bmc_accessibility(self, *, args: str) -> CheckEntry:
        """
        Check BMC accessibility
        :param args: bmc accessibility check specific parameters and arguments
        :return:
        """
        return self._validate_bmc_check(self.ACCESSIBLE)

    def _bmc_stonith(self, *, args: str) -> Union[CheckEntry,
                                                  List[CheckEntry]]:
        """
        Check BMC Stonith Configuration
        :param args: bmc stonith configuration check specific
                     parameters and arguments
        :return:
        """
        return self._validate_bmc_check(self.STONITH)
```
But you can make static this method
```
   @staticmethod
    def _validate_bmc_check(check: str):

        if check == self.STONITH:
            bmc_check = cfg.Checks.BMC_STONITH.value
        elif check == self.ACCESSIBLE:
            bmc_check = cfg.Checks.BMC_ACCESSIBILITY.value
        else:
            raise ValueError(f'BMC Check "{check}" is not supported')
```
There is no contradiction here.

@andkononykhin2 correct me if I am wrong

Reply:
If you look around that function you will see only methods marked as static. So I wonder why it was placed here, looks non-consistent. So options:

-  if you want to keep it non-static, rearrange methods of that class
- or you can make it static (I didn't get why you had chosen something different here and are keeping static methods for all other checks)

For the latter you need to refer your checks using class name (like `Check.STONITH`).
One more option here is classmethod, so you could refer to it as `cls.STONITH` but it would be also non-consistent.

>So it will allow to refer to _validate_bmc_check using self. and be independent of class renaming

`self` can be used for staticmethods instead of classname, so no issues here

```
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class AAA:
...   @staticmethod
...   def test_static():
...     print('STATIC')
...   def test(self):
...     self.test_static()
... 
>>> AAA().test()
STATIC
```